### PR TITLE
Ctpdev: print input and class rates in case of big payload

### PR DESCRIPTION
**Pull request by lietava**

### CODE REVIEW

**Comment by lietava** [comment]
@aferrero2707 - printing inputs and classes for fatal

**Comment by lietava** [comment]
@aferrero2707 , reminder, please, have a look. It is almost trivial.

File: Detectors/CTP/reconstruction/src/RawDataDecoder.cxx [hunk 1]
```diff
@@ -17,6 +17,7 @@
 #include "DataFormatsCTP/TriggerOffsetsParam.h"
 #include "CTPReconstruction/RawDataDecoder.h"
 #include "DataFormatsCTP/Configuration.h"
+#include "DetectorsBase/GRPGeomHelper.h"
```
Comment (aferrero2707) [inline Detectors/CTP/reconstruction/src/RawDataDecoder.cxx:22]:
```suggestion
#include <boost/range/adaptor/reversed.hpp>
```

File: Detectors/CTP/reconstruction/src/RawDataDecoder.cxx [hunk 2]
```diff
@@ -319,6 +321,166 @@ int RawDataDecoder::decodeRaw(o2::framework::InputRecord& inputs, std::vector<o2
...
+    if (rates[i]) {
+      ratesmap[rates[i]] = i;
+    }
+    if (ratesC[i]) {
+      ratesmapC[ratesC[i]] = i;
+    }
+  }
+  auto nhb = o2::base::GRPGeomHelper::getNHBFPerTF();
+  std::string message = "Ringing inputs [MHz]:";
+  for (auto const& r : ratesmap) {
```
Comment (aferrero2707) [inline Detectors/CTP/reconstruction/src/RawDataDecoder.cxx]:
```suggestion
  for (auto const& r : boost::adaptors::reverse(ratesmap)) {
```

File: Detectors/CTP/reconstruction/src/RawDataDecoder.cxx [hunk 3]
```diff
@@ -319,6 +321,166 @@ int RawDataDecoder::decodeRaw(o2::framework::InputRecord& inputs, std::vector<o2
...
+    // LOG(error) << r.second;
+    message += " " + o2::ctp::CTPInputsConfiguration::getInputNameFromIndex(r.second + 1) + ":" + std::to_string(r.first / 32. / o2::constants::lhc::LHCOrbitMUS);
+  }
+  std::string messageC = "Ringing classes [MHz]:";
+  for (auto const& r : ratesmapC) {
```
Comment (aferrero2707) [inline Detectors/CTP/reconstruction/src/RawDataDecoder.cxx]:
```suggestion
  for (auto const& r : boost::adaptors::reverse(ratesmapC)) {
```

File: Detectors/CTP/workflow/src/RawDecoderSpec.cxx [hunk 4]
```diff
@@ -120,15 +121,22 @@ void RawDecoderSpec::run(framework::ProcessingContext& ctx)
     }
     if (payloadSize > (size_t)mMaxInputSize) {
       if (mMaxInputSizeFatal) {
-        LOG(fatal) << "Input data size:" << payloadSize;
+        fatal_flag = 1;
+        LOG(error) << "Input data size bigger than threshold:" << mMaxInputSize << " < " << payloadSize << " ecoding TF and exiting.";
```
Comment (aferrero2707) [inline Detectors/CTP/workflow/src/RawDecoderSpec.cxx]:
```suggestion
        LOG(error) << "Input data size bigger than threshold: " << mMaxInputSize << " < " << payloadSize << " ecoding TF and exiting.";
```

**Review by aferrero2707 — CHANGES_REQUESTED** [review]
@lietava I double-checked the code locally and I confirm it works as expected.
I just left a couple of suggested changes that allow to print the inputs and classes in reverse order, wit the highest rates printed first.

**Comment by lietava** [comment]
@shahor02 : please, merge or comment.

**Review by shahor02 — APPROVED** [review]

**Pull request merged**